src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -220,61 +196,36 @@ public static string GetFileHash(string path)
-                // Since string matching isn't always reliable, failing to find a match is okay.
-                // We only want to show an error if a string match finds multiple matches.
-                // If string matching fails to find a match, show all errors that may occur from ArchAndTypeMatches.
-                if (numOfUrlMatches > 1)
-                {
-                    multipleMatchedInstallers.Add(newInstaller);
-                    continue;
-                }
-                else if (numOfUrlMatches == 0)
+                foreach (var newInstaller in installerUpdate.NewInstallers)
                 {
-                    if (numOfArchAndTypeMatches == 0)
+                    // if the installerUpdate does not have a binary or url architecture specified, then just use what is specified in the installer.
+                    Installer existingInstallerMatch = FindInstallerMatch(
+                        newInstaller,
+                        existingInstallers,
+                        installerManifest,
+                        installerUpdate,
+                        ref unmatchedInstallers,
+                        ref multipleMatchedInstallers);
+
+                    // If a match is found, add match to dictionary and remove for list of existingInstallers
+                    if (existingInstallerMatch != null)
                     {
-                        unmatchedInstallers.Add(newInstaller);
+                        installerMatchDict.Add(existingInstallerMatch, newInstaller);
+                        existingInstallers.Remove(existingInstallerMatch);
                     }
-                    else if (numOfArchAndTypeMatches > 1)
+                    else
                     {
-                        multipleMatchedInstallers.Add(newInstaller);
+                        continue;
                     }
                 }
-
-                var matchingExistingInstaller = numOfUrlMatches == 1 ? urlMatch.Single() : numOfArchAndTypeMatches == 1 ? archAndTypeMatch.Single() : null;
-
-                // If we can't find a match in the remaining existing packages, there must be a mismatch between the old manifest and the URLs provided
-                if (matchingExistingInstaller == null)
-                {
-                    continue;
-                }
-                else
-                {
-                    installerMatchDict.Add(matchingExistingInstaller, newInstaller); // add the match to the map.
-                    existingInstallers.Remove(matchingExistingInstaller);
-                }
             }
 
             if (unmatchedInstallers.Any() || multipleMatchedInstallers.Any())
             {
-                throw new InstallerMatchException(multipleMatchedInstallers, unmatchedInstallers);
+                bool isArchitectureOverride = installerUpdateHelpers.Select(x => x.OverrideArchitecture).Any(i => i.HasValue);
```

Comment:
can just be installerUpdateHelpers.Any(x => x.OverrideArchitecture.HasValue)

Reply:
Thanks, changed to simplified suggestion